### PR TITLE
Correct group labels for keyness

### PR DESCRIPTION
**Pull request by koheiw**
Address #1257, but did not add `docname.keyness() <-`.

### CODE REVIEW

**Comment by koheiw** [comment]
Why this is not merged yet?

**Comment by koheiw** [comment]
I cannot see your query. Can you post it again?

File: R/textstat_keyness.R [hunk 1]
```diff
@@ -93,41 +93,51 @@ textstat_keyness.dfm <- function(x, target = 1L,
...
+        target <- target
+    } else {
+        stop("target must be numeric, character or logical")
+    }
     
+    # use original docnames only when there are two documents
+    if (ndoc(x) == 2) {
+        label <- docnames(x)[order(target, decreasing = TRUE)]
+    } else {
+        label <- c("target", "reference")
```
[the remark this answers is not in the document]
Comment (koheiw) [reply R/textstat_keyness.R]: I basically want people to group documents into two (or remove others) beforehand to make sure that they are comparable (in terms of size especially), but I can use document names for the label ('2017-Trump' in the example) when target is only one document (this is imbalanced comparison) if you want.

**Comment by kbenoit** [comment]
Let's say:
- target should always take on the docname of the target document. 
- reference should take on the docname of the reference if a single "document", otherwise, "Reference"

The question of balance is interesting, since I have seen many corpus linguistics applications to check differential usage of a small set against a large baseline corpus. These make the association measures look very skewed (smaller for the large set) but still identify the words used differently. I think we can leave this to the user.

**Comment by koheiw** [comment]
We only have `target`, so how about adding `reference` to make behavior more synthetic? There was a heated discussion recently on this #1269. I think direct comparison '2017-Trump' vs '2001-Bush' will make charts look nicer (and might be better comparison than '2017-Trump' vs all other).

**Comment by kbenoit** [comment]
I think the discussion in #1269 is more about how than about design. The easy solution is to make the input 
```r
textstat_keyness(pwdfm[c("2017-Trump", "2001-Bush"), ])
```

Do we really want to add an argument to perform a function within `textstat_keyness()` for which there is already a perfectly usable function, namely `[.dfm()`?  The reason that `target` exists is because a document is often compared to a larger baseline, and that makes it convenient to set the comparison group, and because without it we would not have the partition required to perform a keyness computation, unless we insisted on `ndoc(x) == 2`.

**Comment by koheiw** [comment]
Yes, we can choose reference document  using`[.dfm()`, but, if we push this logic further, we need to remove `target`, because we can select target using `dfm_group()`. In my view, we should either have both `target` and `reference` or neither. This is the ghost haunting the `textstat_keyness` (and caused us the chronicle problem)

**Comment by kbenoit** [comment]
We have to have a `target` unless we enforce that `x` has just two documents. Not sure what you mean by the "chronicle problem"...?

I'm okay with adding a `reference = NULL` argument where the default behaviour matches the current behaviour of grouping all non-target documents.

I thought about letting `target`, `reference` be document indexes and having this be groupable, but that raises issues about how to handle overlap, and how to handle defaults, so let's just make it single-document for now (and matching the current behaviour of `target`). Anyone who wants to do more should manipulate the input `x` using indexing.

**Comment by koheiw** [comment]
I think the default value `target = 1L` makes sense only when there are two documents, so I propose to set both `target = NULL` and `reference = NULL` by default:

- In a special case, where `x` only has two documents, `target = 1L` and `reference = 2L` unless users specify differently.
- In other (more general) cases, `target` and `reference` need to be specified explicitly. If either is `NULL` or they overlap with each other, it should raise error.

Alternative idea is to use a single logical vector with `TRUE` (target), `FALSE` (reference) and `NA` (exclude), but sounds a bit crazy even to me.

Please don't mind about "chronicle problem". I am only referring to the fact that we have been wrestling with this seemingly simple problem such a long time.

**Comment by kbenoit** [comment]
+1 for "even to me" 🤣 

KISS rule + Hippocratic oath equivalent: keep `target = 1L` and add `reference = NULL`, which can be replaced by an integer or character document index.

I never saw the lack of `reference` as a problem, but am okay with its addition. This issue was originally about the legend text!

**Comment by koheiw** [comment]
If reference, is only takes a value there is little use. Let's fix the label and finish this off.

**Comment by koheiw** [comment]
It's not an elegant design, but has no problem in practice soI will postpone redesigning.

**Pull request merged**